packages/batch/src/BatchProcessor.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Process native partial responses from SQS, Kinesis Data Streams, and DynamoDB
+ */
+import {
+  BasePartialBatchProcessor,
+  BaseRecord,
+  FailureResponse,
+  SuccessResponse,
+} from '.';
```

Comment:
While at this stage using this type of import is harmless, importing from the default `index.ts` could lead to unintended side effects in the future as the codebase grows.

When it comes to internal files referencing themselves, we prefer to scope imports as much as possible, for example:
```ts
import { BasePartialBatchProcessor } from './BasePartialBatchProcessor';
import type { BaseRecord, FailureResponse, SuccessResponse } from './types';
```

By doing this the file imports only and exactly the entities that are needed for it to work. While this might result in us having to change the imports if we change the folder structure, it'll allow us to be intentional about what we are importing.

Importing from the main file (or in general from files that themselves contain multiple exports) could cause unintended side effects that are hard to track & reproduce, for example imagine we have two classes: `ClassA` and `ClassB`. Both of them are re-exported in `index.ts`:
```ts
export * from './ClassA';
export * from './ClassB';
```

Imagine we now have a third file (`otherModule.ts`) where we need to import `ClassA` only. By using the imports from `index.ts` JS will be requiring and importing both since they are both present in `index.ts`. Generally this is fine, however if `ClassB` is expensive to instantiate we would be putting unnecessary overhead. Additionally, depending on the bundling strategy used by our customers, importing unnecessary modules could make [tree-shaking](https://webpack.js.org/guides/tree-shaking/) more challenging or impossible, further increasing the unnecessary overhead.